libraries/chain/database.cpp

```diff
@@ -2283,6 +2285,33 @@ namespace golos { namespace chain {
                             unclaimed_rewards -= claim;
 
                             const auto &voter = get(itr->voter);
+
+                            if (has_hardfork(STEEMIT_HARDFORK_0_19__756)) {
+                                for (auto& dvir : itr->delegator_vote_interest_rates) {
+                                    auto delegator_claim = claim * dvir.interest_rate / STEEMIT_100_PERCENT;
+
+                                    claim -= delegator_claim;
+
+                                    const auto& delegator = get_account(dvir.account);
+
+                                    asset delegator_vesting = create_vesting(delegator, asset(delegator_claim, STEEM_SYMBOL));
+
+                                    auto vdo_itr = vdo_idx.lower_bound(voter.name);
+                                    if (vdo_itr != vdo_idx.end() && vdo_itr->delegatee == voter.name) {
```

Comment:
contents of this if/else is wrong, you'll get balance mistakes.

please double check that all 4 delegation parts contain right values after returning delegation. this parts are interconnected and must be changed mindfully:
1. delegator's delegated vests
2. delegatee's received vests
3. delegation object's vests (if exist)
4. delegation expiration object's vests (if exist)

you must not change 1,2 without altering one of 3/4 too.

Please, check by hand or write proper tests to make sure balances and objects are consistent

Reply:
@zxcat Let's try to talk in preudocode!
Correct algo is:
```
change1();
change2();
if (3) change3();
if (4) change4();
```
Yes?